backend/app/controllers/spree/admin/stock_items_controller.rb

```diff
@@ -11,41 +18,82 @@ def update
       end
 
       def create
-        variant = Variant.find(params[:variant_id])
-        stock_location = StockLocation.find(params[:stock_location_id])
         stock_movement = stock_location.stock_movements.build(stock_movement_params)
         stock_movement.stock_item = stock_location.set_up_stock_item(variant)
 
         if stock_movement.save
           flash[:success] = flash_message_for(stock_movement, :successfully_created)
+          respond_to do |format|
+            format.json { render json: { stock_item: stock_movement.stock_item, message: flash[:success] } }
+            format.html { redirect_back fallback_location: spree.stock_admin_product_url(variant.product) }
+          end
         else
           flash[:error] = Spree.t(:could_not_create_stock_movement)
+          respond_to do |format|
+            format.json do
+              render json: {
+                errors: stock_movement.errors.full_messages + stock_movement.stock_item.errors.full_messages,
+                message: flash[:error]
+              }, status: :unprocessable_entity
+            end
+            format.html { redirect_back fallback_location: spree.stock_admin_product_url(variant.product) }
+          end
         end
-
-        redirect_back fallback_location: spree.stock_admin_product_url(variant.product)
       end
 
       def destroy
```

Comment:
We can remove this method and override the `render_js_for_destroy` method like this.
```
def render_js_for_destroy
  render :destroy
end
```

Also the `html` format is not being used, so we should do this.